k8s/solr/README.md

```diff
@@ -0,0 +1,417 @@
+ * Set the persistent disk's size for SolrCloud storage. The default disk size is
+   `10Gi`.
+ * Set the persistent disk's size for ZooKeeper storage. The default disk size
+   is `5Gi`.
+
+```shell
+export STORAGE_CLASS="standard" # provide your StorageClass name if not "standard"
+export PERSISTENT_SOLR_SIZE="10Gi"
+export PERSISTENT_ZK_SIZE="5Gi"
+```
+#### Creating a namespace in your Kubernetes cluster
+
+If you use a different namespace than `default`, or if the namespace does
+not exist yet, create a new namespace:
+
+```shell
+kubectl create namespace "${NAMESPACE}"
+```
+
+#### Expanding the manifest template
+
+To expand the template, use `helm template`. We recommend that you save the
+expanded manifest file for future updates to your app.
+
+```shell
+helm template "${APP_INSTANCE_NAME}" \
+  --namespace "${NAMESPACE}" \
+  --set solr.image.repo="${IMAGE_SOLR}" \
+  --set solr.image.tag="${TAG}" \
+  --set zookeeper.image="${IMAGE_ZOOKEEPER}:${ZK_TAG}" \
+  --set deployer.image="${IMAGE_DEPLOYER}" \
+  --set persistence.storageClass="${STORAGE_CLASS}" \
+  --set persistence.solr.storageSize="${PERSISTENT_SOLR_SIZE}" \
+  --set persistence.zookeeper.storageSize="${PERSISTENT_ZK_SIZE}" \
+  --set solr.solrPassword="${SOLR_PASSWORD}" \
+  --set solr.zkReadOnlyPassword="${ZK_READONLY_PASSWORD}" \
+  --set solr.zkAdminPassword="${ZK_ADMIN_PASSWORD}" \
+  --set metrics.exporter.enabled="${METRICS_EXPORTER_ENABLED}" \
+  chart/solr > ${APP_INSTANCE_NAME}_manifest.yaml
+```
+
+#### Applying the manifest to your Kubernetes cluster
+
+To apply the manifest to your Kubernetes cluster, use `kubectl`:
+
+```shell
+kubectl apply -f "${APP_INSTANCE_NAME}_manifest.yaml" --namespace "${NAMESPACE}"
+```
+
+### Accessing the SolrCloud UI web console
+
+The deployed service of SolrCloud is of type ClusterIP, so you can reach the SolrCloud Studio UI within a Kubernetes cluster by port-forwarding.
+
+To accomplish this, run the following commands:
+
+```shell
+# Get "solr" user credentials of SolrCloud UI
+SOLR_USER="solr"
+SOLR_PASSWORD=$(kubectl get secret --namespace \
+  ${NAMESPACE} ${APP_INSTANCE_NAME}-solr-secret \
+  -o jsonpath="{.data.solr-password}" | base64 --decode)
+
+echo "username: ${SOLR_USER}"
+echo "password: ${SOLR_PASSWORD}"
+
+# Forward SolrCloud service port to local workspace
+kubectl port-forward svc/${APP_INSTANCE_NAME}-solr-svc --namespace ${NAMESPACE} 8983
+```
+
+After running the commands, visit [http://localhost:8983](http://localhost:8983)
+from a web browser. Login with `solr` credentials.
+
+### Interacting with SolrCloud via `curl`
+> Note: Port-forwarding should be kept running
+
+```shell
+# Check Cluster status
+curl --user ${SOLR_USER}:${SOLR_PASSWORD} "http://localhost:8983/solr/admin/collections?action=CLUSTERSTATUS"
+
+# Create a "my_collection" with 3 shards (1 shard per replica)
+curl --user ${SOLR_USER}:${SOLR_PASSWORD} "http://localhost:8983/solr/admin/collections?action=CREATE&name=my_collection&numShards=3"
```

Comment:
I would set my_collection in a variable as I see you use the same name in different commands. 

Also you can explain just once the var purpose.